go/fix-chain/fixer.go

```diff
@@ -0,0 +1,282 @@
+package fix_chain
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"github.com/google/certificate-transparency/go/x509"
+	"log"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+const HashSize = sha256.Size
```

Comment:
More comments required - everything that's exported needs a comment (or to not be exported), I'll stop calling them out now.